js/main.js

```diff
@@ -154,61 +157,78 @@ class App extends React.Component {
     return pathname;
   };
 
-  addPaneBatched = (pane) => {
-    if (!this._timeoutID) {
-      this._timeoutID = setTimeout(this.processBatchedPanes, 100);
+  // ------------------ //
+  // batched processing //
+  // ------------------ //
+
+  // store pane to be processed
+  const addPaneBatched = (pane) => {
+    if (!_timeoutID.current) {
+      _timeoutID.current = setTimeout(processBatchedPanes, 100);
     }
-    this._pendingPanes.push(pane);
+    _pendingPanes.current.push(pane);
   };
 
-  processBatchedPanes = () => {
-    let newPanes = Object.assign({}, this.state.panes);
-    let newLayout = this.state.layout.slice();
+  // run processing on queue
+  const processBatchedPanes = () => {
+    // wait until app is mounted
+    if (!mounted.current) {
+      // console.log('not processing');
+      _timeoutID.current = setTimeout(processBatchedPanes, 100);
+      return;
+    }
+    let newPanes = Object.assign({}, storeData.panes);
+    let newLayout = storeData.layout.slice();
 
-    this._pendingPanes.forEach((pane) => {
-      this.processPane(pane, newPanes, newLayout);
+    _pendingPanes.current.forEach((pane) => {
+      processPane(pane, newPanes, newLayout);
     });
 
-    this._pendingPanes = [];
-    this._timeoutID = null;
+    _pendingPanes.current = [];
```

Comment:
In theory this code still has a race condition that may drop pane updates. Not sure if `React` introduces any kind of async that could be unstable. We should do
```
pendingPanes = _pendingPanes.current
_pendingPanes.current = []
penddingPanes.forEach(...
```
_just in-case_

Reply:
Very good point. I am pretty sure I came across this bug for an unpublished speed test I once wrote.
I will definitively include this.

Reply:
added a new commit for this change